packages/core/src/_common/base/api_middleware.js

```diff
@@ -1,9 +1,94 @@
+const datadogLogs = require('@datadog/browser-logs').datadogLogs;
+const formatDate = require('@deriv/shared').formatDate;
+const formatTime = require('@deriv/shared').formatTime;
+
+const datadog_client_token_logs = process.env.DATADOG_CLIENT_TOKEN_LOGS ?? '';
+const is_production = process.env.CIRCLE_JOB === 'release_production';
+const is_staging = process.env.CIRCLE_JOB === 'release_staging';
+const datadog_session_sample_rate = +process.env.DATADOG_SESSION_SAMPLE_RATE ?? 10;
+let datadog_version = '';
+let datadog_env = '';
+const REQUESTS = [
+    'authorize',
+    'website_status',
+    'p2p_advert_create',
+    'p2p_advertiser_payment_methods',
+    'p2p_payment_methods',
+    'p2p_advert_list',
+    'p2p_order_create',
+    'p2p_order_info',
+];
+
+if (is_production) {
+    datadog_version = `deriv-app-${process.env.CIRCLE_TAG}`;
+    datadog_env = 'production';
+} else if (is_staging) {
+    datadog_version = `deriv-app-staging-v${formatDate(new Date(), 'YYYYMMDD')}-${formatTime(Date.now(), 'HH:mm')}`;
+    datadog_env = 'staging';
+}
+
+datadogLogs.init({
+    clientToken: datadog_client_token_logs,
+    site: 'datadoghq.com',
+    forwardErrorsToLogs: false,
+    service: 'Dp2p',
+    sessionSampleRate: datadog_session_sample_rate,
+    version: datadog_version,
+    env: datadog_env,
+});
+
 class APIMiddleware {
     constructor(config) {
         this.config = config;
         this.debounced_calls = {};
+        this.addGlobalMethod();
     }
 
+    getRequestType = request => {
+        let request_type;
+
+        REQUESTS.forEach(type => {
+            if (type in request && !request_type) request_type = type;
+        });
+
+        return request_type;
+    };
+
+    log = (measures = []) => {
+        measures?.forEach(measure => {
+            datadogLogs.logger.info(measure.name, {
```

Comment:
add a check to see if datadog_client_token_logs is present and not '' because on testlinks we don't have this variables and a console error is thrown as the token in not valid.